src/Persistent/Implementation.hs

```diff
@@ -0,0 +1,17 @@
+{-# LANGUAGE CPP #-}
```

Comment:
Yes, the PR is big enough and long-running. so I will keep CPP for now, with cabal flags. But both Persist implementations are being compiled every time, the CPP is just to pick which one to use without tons of needless generalizations everywhere.